backend/schema/validate.go

```diff
@@ -709,6 +791,44 @@ func validatePayloadType(n Node, r Type, v *Verb, reqOrResp string) error {
 	return nil
 }
 
+func requireUnitPayloadType(n Node, r Type, v *Verb, reqOrResp string) error {
+	if _, ok := n.(*Unit); !ok {
+		return errorf(r, "ingress verb %s: GET request type %s must have a body of unit not %s", v.Name, r, n)
+
+	}
+	return nil
+}
+
+func validatePathParamsPayloadType(n Node, r Type, v *Verb, reqOrResp string) error {
+	switch t := n.(type) {
+	case *String, *Data, *Unit, *Float, *Int, *Bool, *Map: // Valid HTTP param payload types.
+	case *TypeAlias:
+		// allow aliases of external types
+		if len(t.Metadata) > 0 {
```

Comment:
This should check that the metadata is of the expected type, `MetadataTypeMap`. There's a helper function (that I cannot for the life of me find at the moment) that can extract a concrete type from a list of sum types if it exists.

Reply:
This was part of the original validation logic, I have updated it.

Reply:
Aaah, thanks @stuartwdouglas